src/sys/unix.rs

```diff
@@ -1389,6 +1391,36 @@ pub(crate) fn original_dst_ipv6(fd: Socket) -> io::Result<SockAddr> {
     .map(|(_, addr)| addr)
 }
 
+// TODO: remove this once https://github.com/rust-lang/libc/pull/4287 merges
+#[cfg(all(feature = "all", any(target_os = "illumos", target_os = "solaris")))]
+const IP_BOUND_IF: libc::c_int = 0x41;
```

Comment:
This is a bit of a bummer; I have an upstream PR rust-lang/libc#4287 to add these constants in `libc`, but wanted to test the change without that. I don't have a strong opinion on whether it's important to wait to pick these up from `libc` or if it's okay to just merge with a local definition and use them from libc in the future, but I imagine the maintainers will have an opinion on this.

Reply:
The libc pr is already merged, so let's wait for a release. I really don't want to maintain constants :)

Reply:
Yeah, I'm happy to wait --- this was just intended to allow testing the change temporarily. Once there's a new libc release, I'll pick it up and update this.